ambry-tools/src/integration-test/java/com.github.ambry/clustermap/HelixBootstrapUpgradeToolTest.java

```diff
@@ -209,6 +232,44 @@ private void writeBootstrapOrUpgrade(long expectedResourceCount, boolean forceRe
     verifyResourceCount(testHardwareLayout.getHardwareLayout(), expectedResourceCount);
   }
 
+  /**
+   * Write the layout files out from the constructed in-memory hardware and partition layouts; use the upload cluster config
+   * tool to upload the partition seal states onto Zookeeper; verify that the writable partitions are consistent between the two.
+   * @throws IOException if a file read error is encountered.
+   * @throws JSONException if a JSON parse error is encountered.
+   */
+  private void uploadClusterConfigs() throws Exception {
+    List<PartitionId> writablePartitions = testPartitionLayout.getPartitionLayout().getWritablePartitions(null);
+    Set<String> writableInPartitionLayout = new HashSet<>();
+    writablePartitions.forEach(k -> writableInPartitionLayout.add(k.toPathString()));
+    Utils.writeJsonObjectToFile(zkJson, zkLayoutPath);
+    Utils.writeJsonObjectToFile(testHardwareLayout.getHardwareLayout().toJSONObject(), hardwareLayoutPath);
+    Utils.writeJsonObjectToFile(testPartitionLayout.getPartitionLayout().toJSONObject(), partitionLayoutPath);
+    HelixBootstrapUpgradeUtil.uploadClusterConfigs(hardwareLayoutPath, partitionLayoutPath, zkLayoutPath,
+        CLUSTER_NAME_PREFIX, DEFAULT_MAX_PARTITIONS_PER_RESOURCE, new HelixAdminFactory());
+    // Check writable partitions in each datacenter
+    for (ZkInfo zkInfo : dcsToZkInfo.values()) {
+      ZkClient zkClient = new ZkClient("localhost:" + zkInfo.getPort(), propertyStoreConfig.zkClientSessionTimeoutMs,
+          propertyStoreConfig.zkClientConnectionTimeoutMs, new ZNRecordSerializer());
+      List<String> subscribedPaths = Collections.singletonList(propertyStoreConfig.rootPath);
+      HelixPropertyStore<ZNRecord> propertyStore =
```

Comment:
We write very similar code to set up `HelixPropertyStore` in many places: `HelixBootstrapUpgradeTool`, `HelixPropertyStoreFactory`, `HelixNotifier`, etc. I think a utility method to instantiate `HelixPropertyStore` could help.

Reply:
true. Let me make relevant changes.